crates/control/src/motion/head_motion.rs

```diff
@@ -52,16 +53,20 @@ impl HeadMotion {
 
     pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
         let current_head_angles = context.sensor_data.positions.head;
-        let raw_request = match context.motion_command.head_motion() {
-            Some(HeadMotionCommand::Center) => *context.center_head_position,
-            Some(HeadMotionCommand::LookAround) | Some(HeadMotionCommand::SearchForLostBall) => {
-                *context.look_around
+        let raw_request = if *context.has_ground_contact {
+            match context.motion_command.head_motion() {
+                Some(HeadMotionCommand::Center) => *context.center_head_position,
+                Some(HeadMotionCommand::LookAround)
+                | Some(HeadMotionCommand::SearchForLostBall) => *context.look_around,
+                Some(HeadMotionCommand::LookAt { .. }) => *context.look_at,
+                Some(HeadMotionCommand::Unstiff) => current_head_angles,
+                Some(HeadMotionCommand::ZeroAngles) => Default::default(),
+                None => Default::default(),
             }
-            Some(HeadMotionCommand::LookAt { .. }) => *context.look_at,
-            Some(HeadMotionCommand::Unstiff) => current_head_angles,
-            Some(HeadMotionCommand::ZeroAngles) => Default::default(),
-            None => Default::default(),
+        } else {
+            Default::default()
         };
```

Comment:
I don't like these nested if and match expressions.
Maybe extract the whole thing into a function and early-return the default if there's no ground contact.
Alternatively, extract only the match statement, then you have a simple 
```rs
let raw_request = if ground contact { joints_from_motion() } else { Default::default();
```
or even
```rs
let raw_request = context
    .has_ground_contact
    .then(|| joints_from_motion(motion))
    .unwrap_or_default();

```

Reply:
Done.
I chose the first option, because the second doesn't seem more readable